app/code/Magento/Customer/Controller/Adminhtml/Index/InlineEdit.php

```diff
@@ -124,6 +124,7 @@ public function execute()
         }
 
         foreach (array_keys($postItems) as $customerId) {
+            $postItems[$customerId]['gender'] = isset($postItems[$customerId]['gender']) ? $postItems[$customerId]['gender'] : '';
```

Comment:
I would recommend using `if` statement in this case for better readability.

```
if (!isset($postItems[$customerId]['gender'])) {
    $postItems[$customerId]['gender'] = null;
}
```

I also think that setting the `null` value instead of an empty string should be enough.

Reply:
Hi @dmytro-ch 
Thanks for reviewing this issue.
I have tried below code which you have suggested
`if (!isset($postItems[$customerId]['gender'])) {
    $postItems[$customerId]['gender'] = null;
}`
unit test failures with this condition also.
I have already debugged `app/code/Magento/Customer/view/adminhtml/ui_component/customer_listing.xml` 
this file, but not getting success.
also Check app/code/Magento/Cms/Controller/Adminhtml/Page/InlineEdit.php File 
Line 116 where I found same condition.
I have updated code, please review it,
With current condition Unit Tests are passed.
Please give your thoughts.
Thanks

Reply:
Here I have set gender as an empty string 

`$customer->setGender('')`

when I use null Functionality not working.

`$customer->setGender(null);`